File: 프로그래머스/3/43162. 네트워크/네트워크.py
import sys
from collections import deque

read = sys.stdin.readline

def solution(n, computers):
    answer = 0
    visited = [False] * n

    for idx in range(n):
        if visited[idx]:
            continue

        q = deque([idx])

        while q:
            node = q.popleft()

            if visited[node]:
                continue

            visited[node] = True

#            for i, n in enumerate(computers[node][node:]):
            for i, n in enumerate(computers[node]):
                if visited[i]:
                    continue

                if n == 1:
                    q.append(i)

        answer += 1

    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    computers = computers.map((nodes) => {
        const nodeSet = new Set();
        for(let i = 0; i < nodes.length; i++) {
            if(nodes[i] === 1) {
                nodeSet.add(i);
            }
        }
        
        return nodeSet;
    })
    
    const visited = new Set();
    let answer = 0;
    
    for(let node = 0; node < n; node++){
        if(visited.has(node)) {
            continue;
        }
        visited.add(node);
        const q = [node];
        
        while(q.length > 0) {
            const i = q.pop();
            
            for(const computer of computers[i]) {
                if(visited.has(computer)) {
                    continue;
                }
                
                q.push(computer);
                visited.add(computer);
            }
            
        }
        answer += 1;
    }
    
    return answer;
}